src/primitives/transaction.cpp

```diff
@@ -62,9 +62,14 @@ uint256 CMutableTransaction::GetHash() const
     return SerializeHash(*this, SER_GETHASH, SERIALIZE_TRANSACTION_NO_WITNESS);
 }
 
-void CTransaction::UpdateHash() const
+void CTransaction::SetHash(const uint256 &tx_hash)
 {
-    *const_cast<uint256*>(&hash) = SerializeHash(*this, SER_GETHASH, SERIALIZE_TRANSACTION_NO_WITNESS);
+    *const_cast<uint256*>(&hash) = tx_hash;
```

Comment:
Hmm... this would result in not longer coupling the hash with the data-structure. This can result in difficult states for future changes.
Would a new primitive (maybe `CIncompleteTransaction` or similar) make more sense?